cpp/src/io/orc/reader_impl.cu

```diff
@@ -196,29 +209,26 @@ class metadata {
         selection.emplace_back(&ff.stripes[stripe], nullptr);
         stripe_rows += ff.stripes[stripe].numberOfRows;
       } while (--max_stripe_count > 0 && ++stripe < get_num_stripes());
-      if (row_count < 0) {
-        row_count = (int)stripe_rows;
-      } else {
-        row_count = std::min(row_count, (int)stripe_rows);
-      }
+      row_count = (row_count < 0) ? static_cast<size_type>(stripe_rows)
+                                  : std::min(row_count, static_cast<size_type>(stripe_rows));
     } else {
       row_start = std::max(row_start, 0);
-      if (row_count == -1) {
-        row_count = get_total_rows();
+      if (row_count < 0) {
+        row_count = static_cast<size_type>(std::min<size_t>(get_total_rows(), INT32_MAX));
       }
       CUDF_EXPECTS(row_count >= 0, "Invalid row count");
-      CUDF_EXPECTS(row_start <= get_total_rows(), "Invalid row start");
+      CUDF_EXPECTS(static_cast<size_t>(row_start) <= get_total_rows(), "Invalid row start");
 
-      int stripe_skip_rows = 0;
-      for (int i = 0, count = 0; i < (int)ff.stripes.size(); ++i) {
+      size_type stripe_skip_rows = 0;
+      for (size_t i = 0, count = 0; i < ff.stripes.size(); ++i) {
         count += ff.stripes[i].numberOfRows;
-        if (count > row_start) {
+        if (count > static_cast<size_t>(row_start)) {
```

Comment:
Both same type
```suggestion
        if (count > row_start) {
```

Reply:
Here too: size_t count vs size_type row_start (ultimately, the reader APIs should probably be moved to size_t even if we can't support more than 2G-row columns, but not in this PR)